src/libsync/creds/oauth.cpp

```diff
@@ -210,13 +210,14 @@ void OAuth::finalize(QPointer<QTcpSocket> socket, const QString &accessToken,
                      const QString &refreshToken, const QString &user, const QUrl &messageUrl) {
     if (!_account->davUser().isNull() && user != _account->davUser()) {
         // Connected with the wrong user
-        QString message = tr("<h1>Wrong user</h1>"
-                                "<p>You logged-in with user <em>%1</em>, but must login with user <em>%2</em>.<br>"
-                                "Please log out of %3 in another tab, then <a href='%4'>click here</a> "
-                                "and log in as user %2</p>")
-                                .arg(user, _account->davUser(), Theme::instance()->appNameGUI(),
-                                    authorisationLink().toString(QUrl::FullyEncoded));
-        httpReplyAndClose(socket, "200 OK", message.toUtf8().constData());
+        qCWarning(lcOauth) << "We expected the user" << _account->davUser() << "but the server answered with user" << user;
+        const QString message = tr("<h1>Wrong user</h1>"
+                                   "<p>You logged-in with user <em>%1</em>, but must login with user <em>%2</em>.<br>"
+                                   "Please log out of %3 in another tab, then <a href='%4'>click here</a> "
+                                   "and log in as user %2</p>")
+                                    .arg(user, _account->davUser(), Theme::instance()->appNameGUI(),
+                                        authorisationLink().toString(QUrl::FullyEncoded));
+        httpReplyAndClose(socket, "401 Unauthorized", message.toUtf8().constData());
```

Comment:
401 means the browser should retry with a password.
I think the right code here would be 403